test/e2e/predicates.go

```diff
@@ -190,4 +193,210 @@ var _ = Describe("Predicates E2E Test", func() {
 		checkError(context, err)
 	})
 
+	// This test verifies we don't allow scheduling of jobs in a way that sum of
+	// limits of tasks is greater than machines capacity.
+
+	It("should validates resource limits of tasks that are allowed to run ", func() {
+		context := initTestContext()
+		defer cleanupTestContext(context)
+
+		// The max cpu available across the nodes
+		nodeMaxAllocatableCpu := int64(0)
+		// The max mem available across the nodes
+		nodeMaxAllocatableMem := int64(0)
+		// Map of available cpu in a node
+		nodeToAllocatableMapCpu := make(map[string]int64)
+		// Map of available mem in a node
+		nodeToAllocatableMemMap := make(map[string]int64)
+		nodeList, err := context.kubeclient.CoreV1().Nodes().List(metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		for _, node := range nodeList.Items {
+			nodeReady := false
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == v1.NodeReady && condition.Status == v1.ConditionTrue {
+					nodeReady = true
+					break
+				}
+			}
+
+			if IsMasterNode(&node) {
+				continue
+			}
+			// Skip the unready node.
+			if !nodeReady {
+				continue
+			}
+			// Find allocatable amount of CPU.
+			allocatable, found := node.Status.Allocatable[v1.ResourceCPU]
+			Expect(found).To(Equal(true))
+			nodeToAllocatableMapCpu[node.Name] = allocatable.MilliValue()
+			if nodeMaxAllocatableCpu < allocatable.MilliValue() {
+				nodeMaxAllocatableCpu = allocatable.MilliValue()
+			}
+
+			// Find allocatable amount of Memory.
+			allocatableMem, found := node.Status.Allocatable[v1.ResourceMemory]
+			Expect(found).To(Equal(true))
+			nodeToAllocatableMemMap[node.Name] = allocatableMem.Value()
+			if nodeMaxAllocatableMem < allocatableMem.Value() {
+				nodeMaxAllocatableMem = allocatableMem.Value()
+			}
+		}
+
+		pods, err := context.kubeclient.CoreV1().Pods(metav1.NamespaceAll).List(metav1.ListOptions{})
+		checkError(context, err)
+		for _, pod := range pods.Items {
+			_, found := nodeToAllocatableMapCpu[pod.Spec.NodeName]
+			_, foundMem := nodeToAllocatableMemMap[pod.Spec.NodeName]
+			if found && foundMem && pod.Status.Phase != v1.PodSucceeded && pod.Status.Phase != v1.PodFailed {
+				nodeToAllocatableMapCpu[pod.Spec.NodeName] -= getRequestedCPU(pod)
```

Comment:
Maybe we could use `GetPodResourceRequest(pod)` to get pod resource request?

Reply:
```GetPodResourceRequest``` method return the max of the pod resource.
We need the sum of the resource in this test case.